src/Value.php

```diff
@@ -138,4 +142,23 @@ public function asNonEmptyStrings(): array
 
         return array_values($this->value);
     }
+
+    public function asDateTimeImmutable(?string $format = null): DateTimeImmutable
+    {
+        $value = $this->asNonEmptyString();
+
+        try {
+            $date = $format !== null
+                ? DateTimeImmutable::createFromFormat($format, $value)
+                : new DateTimeImmutable($value);
+        } catch (Exception $e) {
+            throw new TypeError(sprintf('"%s" is not a valid date string.', $value), 412, $e);
```

Comment:
What's this 412 about?

Reply:
The code was required. So I thought a 412 does fit better than 0

Reply:
Okay, let's keep it for now, i think this will get replaced by PSL anyway